pkg/template/core/starlark_struct.go

```diff
@@ -49,7 +49,11 @@ func (s *StarlarkStruct) AttrNames() []string {
 }
 
 func (s *StarlarkStruct) Get(key starlark.Value) (val starlark.Value, found bool, err error) {
-	if attr, ok := NewStarlarkValue(key).AsGoValue().(string); ok {
+	goValue, err := NewStarlarkValue(key).AsGoValue()
```

Comment:
i would name this attrVal or keyVal

Reply:
we got AsStringValue which seems appropriate to use here. that should eliminate custom error on the next line.

Reply:
👍🏼  ...also pulled type information from the `starlark.Value`.